pygitguardian/utils/tools.py

```diff
@@ -0,0 +1,20 @@
+from typing import Any, Mapping, Sequence
+
+
+def dict_filter_none(dct: Mapping[Any, Any]) -> dict:
+    """Filter a dict to remove all items where the value is None"""
+    return {k: v for k, v in dct.items() if v is not None}
+
+
+def count_not_none(seq: Sequence[Any]) -> int:
+    """Count the number of None values in a sequence"""
+    return sum(0 if v is None else 1 for v in seq)
```

Comment:
nit: can be slightly simplified:
```suggestion
    return sum(1 for v in seq if v is not None)
```